src/Chamilo/UserBundle/Entity/User.php

```diff
@@ -2469,4 +2483,107 @@ public function getSessionAsGeneralCoach()
     {
         return $this->sessionAsGeneralCoach;
     }
+
+    /**
+     * Get a list of properties in this entity
+     * @param EntityManager $em
+     * @param int $recursionDepth Up to how deep we want to get the data
+     * @return array
+     */
+    protected function _toArray($em, $recursionDepth = 0)
+    {
+        // This method is borrowed from https://github.com/borisguery/bgylibrary/blob/master/library/Bgy/Doctrine/EntitySerializer.php
```

Comment:
If you need to pass the EntityManager, it's probably better to put it in the Repository instead of the entity.
The entity should be kept as a "simple" class.

In the User repository you don't need to pass the $em just something like that:

```
class UserRepository extends EntityRepository
{
    public function getUserEntityToArray($recursionDepth) 
   {
        $em = $this->getEntityManager()
        ....
```

Optional bonus:

http://symfony.com/doc/current/components/serializer.html

Reply:
Could you do that? I've published the branch in the project so you can work on it.